content/6_Advanced/Extend_Euclid.mdx

```diff
@@ -221,22 +293,50 @@ $n^2 < \texttt{LLONG\_MAX}$.
 
 ```cpp
 ll invGeneral(ll a, ll b) {
-	array<ll,3> x = extendEuclid(a,b);
-	assert(x[2] == 1); // gcd must be 1
-	return x[0]+(x[0]<0)*b;
+	array<ll, 3> x = extendEuclid(a, b);
+	assert(x[2] == 1);  // gcd must be 1
+	return x[0] + (x[0] < 0) * b;
 }
 
 int main() {
-	FOR(b,1,101) F0R(a,101) if (__gcd(a,b) == 1) {
-		ll x = invGeneral(a,b);
-		assert((a*x-1) % b == 0);
-		assert(0 <= x && x < b);
+	for (int b = 1; b <= 100; b++) {
+		for (int a = 1; a <= 100; a++) {
+			if (__gcd(a, b) == 1) {
+				ll x = invGeneral(a, b);
+				assert((a * x - 1) % b == 0);
+				assert(0 <= x && x < b);
+			}
+		}
 	}
 }
-
 ```
 
 </CPPSection>
+
+<JavaSection>
+
+```java
+static long invGeneral(long a, long b) {
+	long[] x = extendEuclid(a, b);
+	assert (x[2] == 1); // gcd must be 1
+	return x[0] + (x[0] < 0 ? 1 : 0) * b;
+}
+
+public static void main(String[] args) {
```

Comment:
given that there's a main, i'd just put this in a class

Reply:
i'm just gonna remove the main function -> i don't think a test function is necessary